### PR TITLE
Document roles, role inference and role annotations

### PR DESCRIPTION
**Pull request by kl0tl**

### CODE REVIEW

File: language/Roles.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+newtype HTML = MkHTML String
+instance showHTML :: Show HTML where
+  show (MkHTML s) = "(HTML " <> show s <> ")"
+
+shownHTML :: Shown HTML
+shownHTML = Shown (\f -> f (MkHTML "Hello"))
+
+badShownHTML :: Shown HTML
+badShownHTML = coerce shownString
```
Comment (JordanMartinez) [inline language/Roles.md:43]: What is `shownString` in this context?

File: language/Roles.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+```
+
+```
+> shown shownHTML
+"(HTML \"Hello\")"
+
+> shown badShownHTML
+"\"Hello\""
+```
```
Comment (JordanMartinez) [inline language/Roles.md:52]: Perhaps add the following before each example?
```
> :t shownHtml
forall a. Shown a

> :t badShownHtml
forall a. Shown a

**Review by JordanMartinez — COMMENTED** [review]
Just some initial thoughts.

File: language/Roles.md (hunk 1 above)
Comment (kl0tl) [reply language/Roles.md:46]: Oups, I forgot to declare it earlier 😬

**Review by hdgarrood — APPROVED** [review]

**Pull request merged**